core/src/data/geoJsonSource.cpp

```diff
@@ -18,14 +19,17 @@ GeoJsonSource::GeoJsonSource(const std::string& _name, const std::string& _urlTe
     DataSource(_name, _urlTemplate) {
 }
 
-std::shared_ptr<TileData> GeoJsonSource::parse(const Tile& _tile, std::vector<char>& _rawData) const {
+std::shared_ptr<TileData> GeoJsonSource::parse(const TileTask& _task,
+                                               const MapProjection& _projection) const {
+
+    auto& task = dynamic_cast<const DownloadTileTask&>(_task);
```

Comment:
I think you can safely do `static_cast` here also. http://en.cppreference.com/w/cpp/language/static_cast.

Similarly in `mvtSource.cpp` and in ` bench/src/tileLoading.cpp`.